ocp_resources/resource.py

```diff
@@ -1053,36 +1056,44 @@ def _apply_patches(patches, action_text, action):
                 "ResourceEdit <action_text> for resource <resource name>"
                 will be printed for each resource; see below
         """
-        for resource, patch in patches.items():
-            LOGGER.info(
-                f"ResourceEdits: {action_text} data for "
-                f"resource {resource.kind} {resource.name}"
-            )
-
-            # add name to patch
-            if "metadata" not in patch:
-                patch["metadata"] = {}
 
-            # the api requires this field to be present in a yaml patch for
-            # some resource kinds even if it is not changed
-            if "name" not in patch["metadata"]:
-                patch["metadata"]["name"] = resource.name
-
-            if action == "update":
-                resource.update(resource_dict=patch)  # update the resource
+        def _patch():
```

Comment:
This is not needed.
Just create a function `_apply_patches_sampler` and call it instead of `_apply_patches`.

```python
def _apply_patches_sampler(patches, action_text, action):
    return Resource._retry_cluster_exceptions(
            func=_apply_patches, exception_dict=exceptions_dict, patches=patches, action_text=action_text, action=action
        )

you will need to add kwargs to `_retry_cluster_exceptions`
```

Reply:
Ok, I will implement it this way.